src/lib/x509/x509_ext.h

```diff
@@ -500,6 +500,66 @@ class OCSP_NoCheck final : public Certificate_Extension {
       void decode_inner(const std::vector<uint8_t>&) override;
 };
 
+/**
+* TNAuthList extension
+*
+* RFC8226 Secure Telephone Identity Credentials
+*   https://www.rfc-editor.org/rfc/rfc8226#section-9
+*/
+class BOTAN_PUBLIC_API(3, 6) TNAuthList final : public Certificate_Extension {
+   public:
+      class TNEntry final : public ASN1_Object {
+         public:
+            enum Type { TN_ServiceProviderCode = 0, TN_TelephoneNumberRange, TN_TelephoneNumber };
+
+            struct TelephoneNumberRange {
+                  ASN1_String start;  //TelephoneNumber (IA5String)
+                  size_t count;       //2..MAX
+            };
+
+            TNEntry() = default;
+            virtual ~TNEntry() = default;
+
+            void encode_into(DER_Encoder&) const override;
+            void decode_from(class BER_Decoder& from) override;
+
+            Type get_type() const { return type; }
+
+            const std::string& getServiceProviderCode() const { return spc.value(); }
+
+            const std::vector<TelephoneNumberRange>& getTelephoneNumberRange() const { return range; }
+
+            const std::string& getTelephoneNumber() const { return one.value(); }
+
+         private:
+            Type type;
+            ASN1_String spc;                          //ServiceProviderCode
+            std::vector<TelephoneNumberRange> range;  //TelephoneNumberRange
+            ASN1_String one;                          //TelephoneNumber
+      };
+
+      TNAuthList() = default;
+
+      std::unique_ptr<Certificate_Extension> copy() const override { return std::make_unique<TNAuthList>(*this); }
+
+      static OID static_oid() { return OID("1.3.6.1.5.5.7.1.26"); }
```

Comment:
Please register this OID in the `oids.txt` and `oid_maps.cpp`. Here's how to do that:

Add the new OID to `oids.txt`, like so:

```diff

diff --git a/src/build-data/oids.txt b/src/build-data/oids.txt
index 781efa928..6facb03a6 100644
--- a/src/build-data/oids.txt
+++ b/src/build-data/oids.txt
@@ -317,7 +317,9 @@
 2.5.29.35 = X509v3.AuthorityKeyIdentifier
 2.5.29.36 = X509v3.PolicyConstraints
 2.5.29.37 = X509v3.ExtendedKeyUsage
+
 1.3.6.1.5.5.7.1.1 = PKIX.AuthorityInformationAccess
+1.3.6.1.5.5.7.1.26 = PKIX.TNAuthList

 2.5.29.32.0 = X509v3.AnyPolicy
```

Then run this script to auto-generate `oid_maps.cpp` from `oids.txt`. (Perhaps take note of https://github.com/randombit/botan/pull/4120)

`python3 src/scripts/dev_tools/gen_oids.py oids > src/lib/asn1/oid_maps.cpp`